src/source/Signaling/Signaling.c

```diff
@@ -629,6 +680,55 @@ STATUS validateSignalingClientInfo(PSignalingClient pSignalingClient, PSignaling
     return retStatus;
 }
 
+STATUS validateMediaStorageConfig(PSignalingClient pSignalingClient)
+{
+    ENTERS();
+    STATUS retStatus = STATUS_SUCCESS;
+    PSignalingChannelDescription pChannelDescription = NULL;
+    PMediaStorageConfig PMediaStorageConfig = NULL;
+    UINT32 channelNameLen, channelArnLen, storageStramArnLen;
+
+    CHK(pSignalingClient != NULL, STATUS_NULL_ARG);
+    PMediaStorageConfig = &pSignalingClient->mediaStorageConfig;
+    pChannelDescription = &pSignalingClient->channelDescription;
+
+    CHK(channelNameLen = STRNLEN(pChannelDescription->channelName, MAX_CHANNEL_NAME_LEN + 1) <= MAX_CHANNEL_NAME_LEN,
+        STATUS_SIGNALING_INVALID_CHANNEL_NAME_LENGTH);
+    CHK(channelArnLen = STRNLEN(pChannelDescription->channelArn, MAX_ARN_LEN + 1) <= MAX_ARN_LEN, STATUS_SIGNALING_INVALID_CHANNEL_ARN_LENGTH);
+    CHK(STRNCMP(PMediaStorageConfig->channelName, pChannelDescription->channelName, channelNameLen) == 0,
+        STATUS_SIGNALING_MISMATCH_MEDIA_STORAGE_CONFIG);
+
+    if (PMediaStorageConfig->channelArn[0] != '\0') {
```

Comment:
missing a null check for mediastorageconfig

Reply:
We do not need the null check for pMediaStorageConfig since its address comes from pSignalingClient and we added the null check for pSignalingClient.